catalog/app/containers/Errors/ErrorBoundary.tsx

```diff
@@ -1,46 +1,26 @@
 import * as React from 'react'
+import { ErrorBoundary } from 'react-error-boundary'
 import { useLocation } from 'react-router-dom'
+import * as Sentry from '@sentry/react'
 
 import Error from 'components/Error'
 import Layout from 'components/Layout'
-import { createBoundary } from 'utils/ErrorBoundary'
-import * as Sentry from 'utils/Sentry'
-
-interface ErrorBoundaryPlaceholderProps {
-  error: Error
-  info: any
-  reset: () => void
-}
-
-function ErrorBoundaryPlaceholder({ error, info, reset }: ErrorBoundaryPlaceholderProps) {
-  const location = useLocation()
-  const errorShown = React.useRef(false)
-  React.useEffect(() => {
-    if (!errorShown.current) {
-      errorShown.current = true
-      return
-    }
-    errorShown.current = false
-    reset()
-  }, [location.pathname, reset])
-
-  const sentry = Sentry.use()
-  React.useEffect(() => {
-    sentry('captureException', error, info)
-  }, [error, info, sentry])
 
+function FallbackComponent() {
   return (
     <Layout bare>
       <Error headline="Unexpected Error" detail="Something went wrong" />
     </Layout>
   )
 }
 
-const ErrorBoundary = createBoundary(
-  (_: unknown, { reset }: { reset: () => void }) =>
-    (error: $TSFixMe, info: $TSFixMe) => (
-      <ErrorBoundaryPlaceholder error={error} info={info} reset={reset} />
-    ),
-)
+const onError = (error: Error) => Sentry.captureException(error)
 
-export default ErrorBoundary
+export default function AppErrorBoundary({ children }: React.PropsWithChildren<{}>) {
+  const location = useLocation()
+  return (
+    <ErrorBoundary
+      {...{ FallbackComponent, onError, children, resetKeys: [location.pathname] }}
```

Comment:
can this prevent some downstream updates if anything other than pathname changes? e.g. if the route depends on the query string, a query string change should cause a re-rendering of the route, otherwise it may become stuck in an error state